frontend/src/metabase/selectors/app.ts

```diff
@@ -69,9 +69,13 @@ export const getIsAppBarVisible = createSelector(
     embedOptions,
   ) => {
     const isFullscreen = hash.includes("fullscreen");
+    const allTopNavElementsHidden =
+      !embedOptions.search &&
+      !embedOptions.new_button &&
+      !embedOptions.breadcrumbs;
     if (
       !currentUser ||
-      (isEmbedded && !embedOptions.top_nav) ||
+      (isEmbedded && (!embedOptions.top_nav || allTopNavElementsHidden)) ||
```

Comment:
WDYT about extracting this line to a variable or something? IMO it's a bit hard to parse what's going on here

Reply:
that's fair.